### PR TITLE
feat: design system integration part1

### PR DESCRIPTION
**Pull request by abretonc7s**
- integrate UI Provider
- preference provider
- theme provider
- integrate and migrate design system components

### CODE REVIEW

File: packages/sdk-ui/src/components/address-copy-button/address-copy-button.tsx [hunk 1]
```diff
@@ -12,52 +18,65 @@ const sizeMap = {
...
   return (
-    <View style={styles.center}>
-      <IconButton
-        icon="content-copy"
-        size={sizeMap[size]}
-        onPress={handleCopied}
-      />
+    <View style={[styles.container, containerStyle]}>
+      <Pressable style={styles.pressable} onPress={handleCopied}>
+        {showAddress && (
```
Comment (andreahaku) [inline packages/sdk-ui/src/components/address-copy-button/address-copy-button.tsx:50]: if `showAddress` is false we have a button with only the `conten-copy` icon here. Is it ok?

File: packages/sdk-ui/src/components/connected-button/connected-button.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        size={24}
+        color="black"
+      />
+    </View>
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    // padding: 5,
```
Comment (andreahaku) [inline packages/sdk-ui/src/components/connected-button/connected-button.tsx:50]: please, remove commented style

File: packages/sdk-ui/src/components/metamask-button/metamask-button.tsx [hunk 3]
```diff
@@ -13,27 +13,29 @@ const getStyles = () => {
...
-      left: 0,
-      right: 0,
-      bottom: 0,
-      zIndex: 999,
-      backgroundColor: 'transparent', // Transparent background
+      // center modal in screen
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      // position: 'absolute',
```
Comment (andreahaku) [inline packages/sdk-ui/src/components/metamask-button/metamask-button.tsx:27]: please, remove commented style

File: packages/sdk-ui/src/components/network-selector/network-selector.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+          imageSource: t,
+        }}
+        isSelected={chainId === '0x1'}
+        onPress={() => onNetworkChange(mainnetName)}
+        style={styles.networkCell}
+      />
+    );
+  };
+
+  // const renderLineaMainnet = () => {
```
Comment (andreahaku) [inline packages/sdk-ui/src/components/network-selector/network-selector.tsx:43]: why all this commented functions and then the comments in the rendering? Is it not ready?

File: packages/sdk-ui/src/components/sdk-summary-modal/sdk-summary-modal.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  return (
+    <View style={styles.container}>
+      <View style={styles.header}>
+        <Jazzicon size={32} address={account} />
+        <AddressCopyButton address={account ?? ''} />
+        <View style={styles.balanceContainer}>
+          <Text ellipsizeMode="middle" numberOfLines={1}>
+            {balance}
+          </Text>
+          <Text>{chainId === '0x1' ? 'ETH' : '???'}</Text>
```
Comment (andreahaku) [inline packages/sdk-ui/src/components/sdk-summary-modal/sdk-summary-modal.tsx:23]: does this works with mainnet only?

File: packages/sdk-ui/src/components/sdk-summary-modal/sdk-summary-modal.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    </View>
+  );
+};
+
+const styles = StyleSheet.create({
+  container: {
+    display: 'flex',
+    padding: 5,
+    flexDirection: 'column',
+    // flex: 1,
```
Comment (andreahaku) [inline packages/sdk-ui/src/components/sdk-summary-modal/sdk-summary-modal.tsx:41]: please, remove commented styles

File: packages/sdk-ui/src/context/ui-provider.tsx [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+export const WithPreferences = ({
+  children,
+}: {
+  children: React.ReactNode;
+}) => {
+  const preferences = usePreferences();
+  return (
+    <PaperProvider
+      // theme={preferences.theme}
```
Comment (andreahaku) [inline packages/sdk-ui/src/context/ui-provider.tsx:21]: do we need this?

**Comment by abretonc7s** [comment]
@SocketSecurity ignore-all

File: packages/sdk-ui/src/design-system/components/Avatars/Avatar/Avatar.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+} from './variants/AvatarNetwork/AvatarNetwork.stories';
+import AvatarTokenStory, {
+  getAvatarTokenStoryProps,
+} from './variants/AvatarToken/AvatarToken.stories';
+
+// Internal dependencies.
+import { AvatarVariant, AvatarProps } from './Avatar.types';
+import Avatar from './Avatar';
+
+export const getAvatarStoryProps = (): AvatarProps => {
```
Comment (andreahaku) [inline packages/sdk-ui/src/design-system/components/Avatars/Avatar/Avatar.stories.tsx:30]: probably a more concise way to do it is:
```
export const getAvatarStoryProps = (): AvatarProps => {
  const avatarVariant = select(
    'Avatar Variant',
    AvatarVariant,
    AvatarVariant.Account,
    storybookPropsGroupID,
  );

  const avatarPropsGetters = {
    [AvatarVariant.Account]: getAvatarAccountStoryProps,
    [AvatarVariant.Favicon]: getAvatarFaviconStoryProps,
    [AvatarVariant.Icon]: getAvatarIconStoryProps,
    [AvatarVariant.Network]: getAvatarNetworkStoryProps,
    [AvatarVariant.Token]: getAvatarTokenStoryProps,
  };

  return {
    variant: avatarVariant,
    ...avatarPropsGetters[avatarVariant](),
  };
};
```

File: packages/sdk-ui/src/design-system/components/Avatars/Avatar/Avatar.tsx [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+import AvatarAccount from './variants/AvatarAccount';
+import AvatarFavicon from './variants/AvatarFavicon';
+import AvatarIcon from './variants/AvatarIcon';
+import AvatarNetwork from './variants/AvatarNetwork';
+import AvatarToken from './variants/AvatarToken';
+
+// Internal dependencies.
+import { AvatarProps, AvatarVariant } from './Avatar.types';
+
+const Avatar = (avatarProps: AvatarProps) => {
```
Comment (andreahaku) [inline packages/sdk-ui/src/design-system/components/Avatars/Avatar/Avatar.tsx:13]: similarly here:
```
const Avatar = (avatarProps: AvatarProps) => {
  const AvatarComponents = {
    [AvatarVariant.Account]: AvatarAccount,
    [AvatarVariant.Favicon]: AvatarFavicon,
    [AvatarVariant.Icon]: AvatarIcon,
    [AvatarVariant.Network]: AvatarNetwork,
    [AvatarVariant.Token]: AvatarToken,
  };

  const SelectedAvatar = AvatarComponents[avatarProps.variant];

  if (!SelectedAvatar) {
    throw new Error('Invalid Avatar Variant');
  }

  return <SelectedAvatar {...avatarProps} />;
};
```

File: packages/sdk-ui/src/design-system/components/Avatars/AvatarGroup/AvatarGroup.constants.ts [hunk 10]
```diff
@@ -0,0 +1,94 @@
+// Internal dependencies.
+import { AvatarGroupTokenList } from './AvatarGroup.types';
+
+export const AVAILABLE_TOKEN_LIST: AvatarGroupTokenList = [
+  {
```
Comment (andreahaku) [inline packages/sdk-ui/src/design-system/components/Avatars/AvatarGroup/AvatarGroup.constants.ts:5]: all the same? are they placeholders for now?

File: packages/sdk-ui/src/design-system/components/Avatars/AvatarGroup/AvatarGroup.tsx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+          );
+        }),
+    [tokenList, avatarSpacing, extraSmallSize, styles.stackedAvatarWrapper],
+  );
+
+  return (
+    <View style={styles.base}>
+      <View style={styles.stack}>{renderTokenList()}</View>
+      <View style={styles.overflowCounterWrapper}>
+        {shouldRenderOverflowCounter && (
```
Comment (andreahaku) [inline packages/sdk-ui/src/design-system/components/Avatars/AvatarGroup/AvatarGroup.tsx:61]: why don't you add `shouldRenderOverflowCounter &&` to the wrapping <view>

**Review by andreahaku — APPROVED** [review]
a few comments. feel free to ignore them for this first version of the integration

**Pull request merged**